z3tracer/src/model.rs

```diff
@@ -73,7 +72,7 @@ pub struct Model {
     // Quantifier instantiations indexed by (hexadecimal) key.
     instantiations: BTreeMap<QIKey, QuantInstantiation>,
     // Stack of current quantifier instances.
-    current_instances: Vec<(QIKey, QuantInstantiationData)>,
+    current_instances: Vec<(QIKey, QuantInstantiationData, usize)>,
```

Comment:
nit: Because `usize` is not self-explanatory, at this point, I would generally create a struct with named fields for `(QIKey, QuantInstantiationData, usize)`.